craft_parts/executor/part_handler.py

```diff
@@ -418,7 +418,7 @@ def _run_prime(
             and is_deb_based()
         ):
             primed_stage_packages = _get_primed_stage_packages(
-                contents.files, prime_dir=self._part.prime_dir
+                contents.files, prime_dir=self._part.base_prime_dir
```

Comment:
This is something we didn't think about, should primed packages list be global or per partition? My gut feeling is that we should do it per partition, and if global is more appropriate the application could merge the lists.